pkg/cli/extract/extract.go

```diff
@@ -105,19 +105,24 @@ func (o *ExtractOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, args [
 		return err
 	}
 
+	if o.TargetDirectory != "-" {
```

Comment:
I'd prefer to move creation of the target directory as close to the line of code where it is needed. E.g. https://github.com/openshift/oc/blob/master/pkg/cli/extract/extract.go#L167

Reply:
When creating the target directory right before it is needed one might get "permission denied". Knowing this in advance would help to save the entire extraction step on the other hand. Which makes the current change more preferable on the other hand.